src/hooks/use-fetch-wpcom-sites/index.tsx

```diff
@@ -156,7 +141,34 @@ export const useFetchWpComSites = ( connectedSiteIds: number[] ) => {
 				}
 			);
 
-			isInitialized.current = true;
+			const syncSites = transformSiteResponse(
+				response.sites,
+				allConnectedSites.map( ( { id } ) => id )
+			);
```

Comment:
```suggestion
			const syncSites = transformSiteResponse( response.sites, [] );
```
Since we're always running `transformSiteResponse` for `syncSitesWithSyncSupportForSelectedSite` anyway, I believe we could just pass an empty array here, right? It doesn't matter what value we store for `syncSupport` in `appdata`.

Reply:
Maybe this is actually a decent enough simplification to not move the `already-connected` logic elsewhere… Though I still find it very tempting since that's the only non-global sync status.

Reply:
`It doesn't matter what value we store for syncSupport in appdata.` - hm, maybe you are right,  but I want to keep it appdata-v1.json with correct syncStatus, to avoid confusing us in the future with not aligned data between appdata-v1.json and what we have in runtime.

Reply:
AFAICT, all `syncStatus` keys in appdata already have the value `already-connected`, except for after they're updated by this call:
https://github.com/Automattic/studio/blob/ce42b2ae0e31b5f680145f7714a56728feb8e645/src/hooks/sync-sites/sync-sites-context.tsx#L65

I would argue it's better to apply my suggestion here. It will make the `syncStatus` value more accurate in appdata.

Footnote: <del>I noticed that I had some lingering connected sites in my appdata file with `localSiteId` values of deleted sites. I'll look into this separately to ensure we clean up `connectedWpcomSites` when local sites are deleted.</del> <ins>Nevermind, those were just connected sites for another user.</ins>

Reply:
I just tested this out, and it turns out this doesn't work as easily as I'd hoped. Let's skip this for now and follow up in https://github.com/Automattic/dotcom-forge/issues/10156